test/select.jl

```diff
@@ -0,0 +1,838 @@
+    select!(d, Not(r"b"))
+    @test d == DataFrame(d=4)
+
+    d = copy(df)
+    select!(d, Not([2, 5, 3]))
+    @test names(d) == [:a, :d]
+    select!(d, Not(2))
+    @test d == DataFrame(a=1)
+
+    d = copy(df)
+    select!(d, Not(2:3))
+    @test d == DataFrame(a=1, d=4, e=5)
+
+    d = copy(df)
+    select!(d, Not([false, true, true, false, false]))
+    @test d == DataFrame(a=1, d=4, e=5)
+end
+
+@testset "select Not" begin
+    df = DataFrame(a=1, b=2, c=3, d=4, e=5)
+    @test_throws BoundsError select(df, Not(0))
+    @test_throws BoundsError select(df, Not(6))
+    @test_throws ArgumentError select(df, Not([1, 1]))
+    @test_throws ArgumentError select(df, Not(:f))
+    @test_throws BoundsError select(df, Not([true, false]))
+
+    df2 = copy(df)
+    d = select(df, Not([:a, :e, :c]))
+    @test names(d) == [:b, :d]
+    @test d == df[:, [:b, :d]]
+    @test d.b !== df.b
+    @test d.d !== df.d
+    @test df == df2
+
+    df2 = copy(df)
+    d = select(df, Not(r"[aec]"))
+    @test names(d) == [:b, :d]
+    @test d == df[:, [:b, :d]]
+    @test d == df[:, r"[bd]"]
+    @test d.b !== df.b
+    @test d.d !== df.d
+    @test df == df2
+
+    d = select(df, Not([2, 5, 3]))
+    @test names(d) == [:a, :d]
+    @test d.a !== df.a
+    @test d.d !== df.d
+    @test d == df[:, [:a, :d]]
+    @test df == df2
+
+    d = select(df, Not(2:3))
+    @test d == DataFrame(a=1, d=4, e=5)
+    @test d.a !== df.a
+    @test d.d !== df.d
+    @test d.e !== df.e
+    @test df == df2
+
+    d = select(df, Not([false, true, true, false, false]))
+    @test d == DataFrame(a=1, d=4, e=5)
+    @test d.a !== df.a
+    @test d.d !== df.d
+    @test d.e !== df.e
+    @test df == df2
+
+    d = select(df, Not(1))
+    @test d == DataFrame(b=2,c=3,d=4,e=5)
+    @test d.b !== df.b
+    @test d.b == df.b
+    @test df == df2
+
+    d = select(df, Not([:a, :e, :c]), copycols=false)
+    @test names(d) == [:b, :d]
+    @test d == df[:, [:b, :d]]
+    @test d.b === df.b
+    @test d.d === df.d
+    @test df == df2
+
+    d = select(df, Not(r"[aec]"), copycols=false)
+    @test names(d) == [:b, :d]
+    @test d == df[:, [:b, :d]]
+    @test d == df[:, r"[bd]"]
+    @test d.b === df.b
+    @test d.d === df.d
+    @test df == df2
+
+    d = select(df, Not([2, 5, 3]), copycols=false)
+    @test names(d) == [:a, :d]
+    @test d.a === df.a
+    @test d.d === df.d
+    @test d == df[:, [:a, :d]]
+    @test df == df2
+
+    d = select(df, Not(2:3), copycols=false)
+    @test d == DataFrame(a=1, d=4, e=5)
+    @test d.a === df.a
+    @test d.d === df.d
+    @test d.e === df.e
+    @test df == df2
+
+    d = select(df, Not([false, true, true, false, false]), copycols=false)
+    @test d == DataFrame(a=1, d=4, e=5)
+    @test d.a === df.a
+    @test d.d === df.d
+    @test d.e === df.e
+    @test df == df2
+
+    d = select(df, Not(1), copycols=false)
+    @test d == DataFrame(b=2,c=3,d=4,e=5)
+    @test d.b === df.b
+    @test df == df2
+end
+
+@testset "select Not view" begin
```

Comment:
```suggestion
@testset "select Not on SubDataFrame" begin
```
Couldn't this use the same block as above with a `for` loop?

Reply:
well - we probably could write some ifs inside. The difference is that with `SubDataFrame` we have to do additional checks on type of return value of the operation. I will clean up these tests, but as we are just moving old tests to a new place I do not want to change them too much.